src/JestExt.ts

```diff
@@ -562,7 +459,19 @@ export class JestExt {
       }
     })
 
-    vscode.debug.startDebugging(vscode.workspace.workspaceFolders[0], configuration)
+    const workspaceFolder = vscode.workspace.workspaceFolders[0]
+    try {
+      // try to run the debug configuration from launch.json
+      await vscode.debug.startDebugging(workspaceFolder, 'vscode-jest-tests')
+    } catch {
+      // if that fails, there (probably) isn't any debug configuration (at least no correctly named one)
+      vscode.window.showWarningMessage(
+        'Debug Jest test: No correctly named debug configuration found in launch.json, starting debugging using the default configuration.'
+      )
+      // therefore run the test using the default configuration
+      const debugConfiguration = this.snippetDebugConfigurationProvider.provideDebugConfigurations(workspaceFolder)[0]
+      await vscode.debug.startDebugging(workspaceFolder, debugConfiguration)

```

Comment:
this `startDebugging` could fail because the generated config is not correct, for example, the program or runtimeExecutable is wrong... therefore, a good place to catch and instruct users how to examine and customize their debug config.

Reply:
You mean the "fallback" `startDebugging`? The issue is that even in such cases `startDebugging` resolves the Promise successfully, after automatically displaying an error message (`The attribute "runtimeExecutable" doesn't exist ("/path/to/jest").`). One therefore has to initiate a change of the VS Code API or show the user a warning while falling back to the default configuration or just let VS Code show its error message.